Quest 데이터 테이블 추가 및 QuestManager 수정

SLSL-SL opened the pull request:
quest_list.json : 퀘스트 데이터 테이블 (어떤 내용의 퀘스트가 있는지 적힌 메뉴판) 기존 quest.json과 다른 경로로 저장. (외부에서 작성한 파일이기 때문)
위의 변경 사항으로 인한 QuestManager 수정

SLSL-SL committed:
Quest 데이터 테이블 추가 및 QuestManager 수정

quest_list.json : 퀘스트 데이터 테이블 (어떤 내용의 퀘스트가 있는지 적힌 메뉴판)
기존 quest.json과 다른 경로로 저장. (외부에서 작성한 파일이기 때문)
위의 변경 사항으로 인한 QuestManager 수정

SLSL-SL commented:
오류 메세지 발견 후 확인을 위해 닫았다가 재활성화 된 풀 리퀘스트

ChoiSeonMun commented on Assets/Scripts/QuestDialog.cs:
Quest 도메인을 따로 설계할 것

ChoiSeonMun commented on Assets/Scripts/QuestManager.cs:
2000~2010년대 옛날 게임회사 식 코드

ChoiSeonMun replied:
컴포넌트일 이유가 없음

ChoiSeonMun commented on Assets/StreamingAssets/quest_list.json:
도메인 무결성 지켜라..

SLSL-SL committed:
타일맵 업데이트

타일맵 외곽 범위 확장 배치

SLSL-SL committed:
QuestSystem 리팩터링을 통한 DDD기반 설계로 수정 작업

기존의 QuestData, Dialog 등을 삭제하고 Manager의 경우 DDD 기반 설계로 수정.
DTO 추가, 보상 범위값 구현을 위한 스크립트 추가

이후 서비스 스크립트 구현 예정

SLSL-SL committed:
QuestSystem DDD기반 설계 -2-

보상 지급을 위한 QuestRewardService 구현.

GDKHJ reviewed (approved):
일단 병합 후 리팩토링 진행하겠습니다.
